### DISCUSSION TITLE
Issue with custom_lint_client dependency resolution for private repositories

### DISCUSSION THREAD

**Discussion by MobiliteDev** [invertase/dart_custom_lint#285] [Q&A] [answered]
Hello,

EDIT: I've seen that the section "I have a problem and I need help" that you access when creating an issue redirect to riverpod's repo discussion section so I don't really know if I should post this here or on riverpod's repo.

I’m experiencing issues when using `custom_lint` in a project that depends on a package (`custom_pedantic`) hosted on a private Unpub repository. The problem occurs specifically when switching from a local path dependency to the hosted version on our private Unpub server.

### Issue Description
In our project, we have a package named `custom_pedantic`, which contains our custom lint rules and is hosted on a private Unpub server. I’ve configured the dependency in `pubspec.yaml` as follows:

```yaml
custom_pedantic:
  hosted:
    name: custom_pedantic
    url: https://unpub.custom.fr/
  version: ^3.1.0
```

However, when `custom_lint` generates the client (`custom_lint_client`), I get the following error:

```
Exception: Failed to run "pub get" in the client project:
Resolving dependencies...

Because custom_lint_client depends on custom_pedantic any which doesn't exist (could not find package custom_pedantic at https://pub.dev), version solving failed.
```

It seems like the generated client (`custom_lint_client`) is trying to fetch the `custom_pedantic` package from `pub.dev` instead of using the hosted URL I provided in `pubspec.yaml` (which is not the case when the same dependency is linked with path).

Is there a way to make it work with private repositories or should i publish my ``custom_pedantic` pacakge to pub.dev so that can work ?

Thanks in advance for any guidance or information on this. I'd be happy to provide more details if needed!

**Comment by rrousselGit** [comment, marked as answer]
Custom_lint doesn't support it at the moment. Feel free to make an issue, but that's low priority for me.
